src/http/representation/RepresentationMetadata.ts

```diff
@@ -304,6 +306,59 @@ export class RepresentationMetadata {
     return this;
   }
 
+  private setContentType(input?: ContentType | string): void {
+    // Make sure complete Content-Type RDF structure is gone
+    this.removeContentType();
+
+    if (!input) {
+      return;
+    }
+
+    if (typeof input === 'string') {
+      input = parseContentType(input);
+    }
+
+    for (const [ key, value ] of Object.entries(input.parameters)) {
+      const node = DataFactory.blankNode();
+      this.addQuad(this.id, SOLID_META.terms.contentTypeParameter, node);
+      this.addQuad(node, RDFS.terms.label, key);
+      this.addQuad(node, SOLID_META.terms.value, value);
+    }
+
+    // Set base content type string
+    this.set(CONTENT_TYPE_TERM, input.value);
+  }
+
+  /**
+   * Parse the internal RDF structure to retrieve the Record with ContentType Parameters.
+   * @returns An Record&lt;string,string&gt; object with Content-Type parameters and their values.
+   */
+  private getContentType(): ContentType | undefined {
+    const value = this.get(CONTENT_TYPE_TERM)?.value;
+    if (!value) {
+      return;
+    }
+    const params = this.getAll(SOLID_META.terms.contentTypeParameter);
+    return {
+      value,
+      parameters: Object.fromEntries(params.map((param): [string, string] => [
+        this.store.getObjects(param, RDFS.terms.label, null)[0].value,
+        this.store.getObjects(param, SOLID_META.terms.value, null)[0].value,
+      ])),
```

Comment:
```suggestion
      parameters: Object.fromEntries(params.map((param): [string, string] => {
        const labels = this.store.getObjects(param, RDFS.terms.label, null);
        const values = this.store.getObjects(param, SOLID_META.terms.value, null);
        if (labels.length !== 1 || values.length !== 1) {
          this.logger.error(`Detected invalid content-type metadata for ${this.id.value}`);
          return [ 'invalid', '' ];
        }
        return [ labels[0], values[0] ];
      })),
```
It should never happen in case the content-type functions are used correctly, but just in case someone manually adds invalid metadata.